web/src/compose_validate.ts

```diff
@@ -78,6 +96,45 @@ export function get_disabled_send_tooltip(): string {
     return "";
 }
 
+export function get_disabled_save_tooltip($elem: JQuery): void {
+    const $button_wrapper = $elem.find<tippy.PopperElement>(".message_edit_save_container");
+    let tippy_instance = util.the($button_wrapper)._tippy;
+
+    if ($button_wrapper.hasClass("times-up")) {
+        // We need to destroy any existing tooltip that may appear
+        // when the message length limit is exceeded during editing.
+        if (tippy_instance) {
+            tippy_instance.destroy();
+        }
+        tippy_instance = util.the($button_wrapper)._tippy;
+    }
+
+    if (message_too_long || $button_wrapper.hasClass("times-up")) {
+        const content = !$button_wrapper.hasClass("times-up")
+            ? $t(
+                  {
+                      defaultMessage: `Message length shouldn't be greater than {max_length} characters.`,
+                  },
+                  {
+                      max_length: realm.max_message_length,
+                  },
+              )
+            : $t({
+                  defaultMessage: `You can no longer save changes to this message.`,
+              });
+
+        if (!tippy_instance) {
+            // avoid duplication of tippy
+            const opts: Partial<tippy.Props> = {placement: "top"};
+            settings_components.initialize_disable_btn_hint_popover($button_wrapper, content, opts);
+        }
+    } else {
+        if (tippy_instance) {
+            tippy_instance.destroy();
+        }
+    }
+}
```

Comment:
I am not sure if this is the best way to dynamically set up the content of the tooltip.
If you guys know any better approach then kindly guide me, I will update my PR accordingly.

Reply:
This seems a bit messy; usually I'd expect logic in the `onShow` function for a tooltip, like this one:

```
    message_list_tooltip(".message-list .message-time", {                                               
        onShow(instance) {                                                                              
            const $time_elem = $(instance.reference);                                                   
            const $row = $time_elem.closest(".message_row");                                            
            assert(message_lists.current !== undefined);                                                
            const message = message_lists.current.get(rows.id($row))!;                                  
            // Don't show time tooltip for locally echoed message.                                      
            if (message.locally_echoed) {                                                               
                return false;                                                                           
            }                                                                                           
            const time = new Date(message.timestamp * 1000);                                            
            instance.setContent(timerender.get_full_datetime_clarification(time));                      
            return undefined;                                                                           
        },                                                                                              
        onHidden(instance) {                                                                            
            instance.destroy();                                                                         
        },                                                                                              
    });                                                                                                 
```